src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/PersonAbsenceTests.cs

```diff
@@ -98,6 +101,110 @@ public async Task DeleteAbsence_ShouldBeOk_WhenAdmin()
             response.Should().BeSuccessfull();
         }
 
+        [Fact]
+        public async Task ShouldGetRoleNameFromBasePositionWhenEmpty()
+        {
+            const string BasePositionName = "Base position name";
+            var testProject = new FusionTestProjectBuilder();
+            var basePosition = testProject
+                .AddBasePosition(BasePositionName);
+            OrgServiceMock.AddProject(testProject);
+
+            var request = new CreatePersonAbsenceRequest
+            {
+                AppliesFrom = new DateTime(2021, 04, 30),
+                AppliesTo = new DateTime(2022, 04, 30),
+                Comment = "A comment",
+                Type = ApiPersonAbsence.ApiAbsenceType.OtherTasks,
+                AbsencePercentage = null,
+                IsPrivate = false,
+                TaskDetails = new ApiTaskDetails
+                {
+                    BasePositionId = basePosition.Id,
+                    TaskName = "task name",
+                    RoleName = null,
+                    Location = "location"
+                }
+            };
+
+            using var authScope = fixture.AdminScope();
+            var response = await client.TestClientPutAsync<TestAbsence>($"/persons/{testUser.AzureUniqueId}/absence/{TestAbsenceId}", request);
+            response.Should().BeSuccessfull();
+
+            response.Value.TaskDetails?.RoleName.Should().Be(BasePositionName);
+        }
+
+        [Fact]
+        public async Task PrivateNotesShouldBeHiddenForOtherResourceOwners()
+        {
+            var siblingResourceOwner = fixture.AddProfile(FusionAccountType.Employee);
+            siblingResourceOwner.FullDepartment = "TPD PRD TST QWE ABC";
+            siblingResourceOwner.Department = "TST QWE ABC";
+            siblingResourceOwner.IsResourceOwner = true;
+
+            var request = new CreatePersonAbsenceRequest
+            {
+                AppliesFrom = new DateTime(2021, 04, 30),
+                AppliesTo = new DateTime(2022, 04, 30),
+                Comment = "A comment",
+                Type = ApiPersonAbsence.ApiAbsenceType.OtherTasks,
+                AbsencePercentage = null,
+                IsPrivate = true,
+
+                TaskDetails = new ApiTaskDetails
+                {
+                    TaskName = "Top secret task name",
+                    RoleName = "Top secret role name",
+                    Location = "Top secret location"
+                }
+            };
+
+            using var authScope = fixture.AdminScope();
```

Comment:
When using multiple scopes, i would prefer to use the using block, as it's more descriptive on which api calls are using elevated access..

```
using (var adminScope = fixture.AdminScope()) 
{
   await POST();
}
// or we can use the 'using var' here.. 
using (var limitedAuthScope = fixture.UserScope(siblingResourceOwner)) 
{
   ....
}
```

The key is that the auth scope should be disposed.
Now it is working because the new scope is overriding the other scope.. But a using block makes it more explicit that this is different than the main scope
}